dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -232,79 +234,124 @@ std::wstring Microsoft::Windows::AppNotifications::Helpers::SetDisplayNameBasedO
-        nullptr /* lpdwDisposition */));
+    return E_NOTIMPL;
+}
 
-    // Retrieve the display name
-    std::wstring displayName{};
-    displayName = SetDisplayNameBasedOnProcessName();
+// Do nothing. This is just a placeholder while the UDK is ingested with the proper API.
+HRESULT ToastNotifications_RetrieveAssets_Stub(_Out_ PWSTR* displayName, _Out_ PWSTR* iconFilePath)
+{
+    *displayName = nullptr;
+    *iconFilePath = nullptr;
+
+    return E_NOTIMPL;
+}
+
+HRESULT RetrieveAssetsFromWindow(_Out_ PWSTR* displayName, _Out_ PWSTR* iconFilePath)
+{
+    *displayName = nullptr;
+    *iconFilePath = nullptr;
 
-    // Retrieve the icon
-    std::wstring iconFilePath{};
+    wil::unique_cotaskmem_string localDisplayName;
+    wil::unique_cotaskmem_string localIconFilePath;
 
-    wil::unique_hwnd hWindow{ GetConsoleWindow() };
+    HWND hWindow = GetForegroundWindow();
 
     if (hWindow)
     {
-        // Retrieve DisplayName and IconUri
-        // - DisplayName: Retrieve from Shell. If not specified, fall back to filename.
-        // - Icon: Retrieve from Shell. If it's not the case or the file extension is unsupported, then throw.
         winrt::com_ptr<IPropertyStore> propertyStore;
-        THROW_IF_FAILED(SHGetPropertyStoreForWindow(hWindow.get(), IID_PPV_ARGS(propertyStore.put())));
+        THROW_IF_FAILED(SHGetPropertyStoreForWindow(hWindow, IID_PPV_ARGS(propertyStore.put())));
 
         wil::unique_prop_variant propVariantDisplayName;
-        // Do not throw in case of failure, default to the filepath approach below as fallback to set a DisplayName.
         THROW_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchDisplayNameResource, &propVariantDisplayName));
 
-        if (propVariantDisplayName.vt == VT_LPWSTR && propVariantDisplayName.pwszVal != nullptr && wcslen(propVariantDisplayName.pwszVal) > 0)
+        if (propVariantDisplayName.vt == VT_LPWSTR && Microsoft::Windows::AppNotifications::Helpers::IsWideStringEmptyOrNull(propVariantDisplayName.pwszVal))
         {
-            displayName = propVariantDisplayName.pwszVal;
+            localDisplayName = wil::make_unique_string<wil::unique_cotaskmem_string>(propVariantDisplayName.pwszVal);
         }
 
         wil::unique_prop_variant propVariantIcon;
         THROW_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchIconResource, &propVariantIcon));
 
-        THROW_HR_IF_MSG(E_UNEXPECTED, (propVariantIcon.vt == VT_EMPTY || propVariantIcon.pwszVal == nullptr), "Icon is not specified");
-
-        THROW_HR_IF_MSG(E_UNEXPECTED, propVariantIcon.vt != VT_LPWSTR, "Icon should be a valid Unicode string");
+        THROW_HR_IF(E_UNEXPECTED, propVariantIcon.vt != VT_LPWSTR || Microsoft::Windows::AppNotifications::Helpers::IsWideStringEmptyOrNull(propVariantIcon.pwszVal));
 
-        THROW_HR_IF_MSG(E_UNEXPECTED, wcslen(propVariantIcon.pwszVal) == 0, "Icon is an empty string");
+        std::wstring localIconFilePathAsWstring = propVariantIcon.pwszVal;
```

Comment:
NIT: `std::wstring localIconFilePathString{ propVariantIcon.pwszVal };`

[ES.23: Prefer the {}-initializer syntax](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#es23-prefer-the--initializer-syntax)